Mathlib/Topology/Algebra/Nonarchimedean/TotallyDisconnected.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2024 Jou Glasheen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jou Glasheen, Kevin Buzzard, David Loeffler, Yongle Hu
+-/
+
+import Mathlib.Topology.Algebra.Nonarchimedean.Basic
+
+/-!
+# Total separatedness of nonarchimedean groups
+
+In this file, we prove that a nonarchimedean group is a totally separated topological space.
+
+## Main results
+
+- `NonarchimedeanGroup.instTotallySeparated`: A nonarchimedean group is a
+                                                      totally separated topological space.
+
+## Notation
+
+ - `G` : Is a nonarchimedean group.
+ - `V` : Is an open subgroup which is a neighbourhood of the identity in `G`.
+
+## References
+
+See Proposition 2.3.9 and Problem 63 in :
+Gouvêa, F. Q. (2020) p-adic Numbers An Introduction. 3rd edition.
+  Cham, Springer International Publishing.
+-/
+
+open Pointwise TopologicalSpace
+
+variable {G : Type*} [TopologicalSpace G] [Group G] [NonarchimedeanGroup G] [T2Space G]
+
+namespace NonarchimedeanGroup.auxiliary
```

Comment:
I don't think this namespace is needed. The lemma can go into the `NonarchimedeanGroup` namespace. (Or you inline it into the instance, as David suggested.)